api/base/serializers.py

```diff
@@ -1358,20 +1358,18 @@ def to_representation(self, obj, envelope='data'):
             context_envelope = None
         enable_esi = self.context.get('enable_esi', False)
         is_anonymous = is_anonymized(self.context['request'])
-        to_be_removed = set()
-        if is_anonymous and hasattr(self, 'non_anonymized_fields'):
-            # Drop any fields that are not specified in the `non_anonymized_fields` variable.
-            allowed = set(self.non_anonymized_fields)
-            existing = set(self.fields.keys())
-            to_be_removed = existing - allowed
+        not_allowed = []
+        if is_anonymous and hasattr(self, 'anonymized_fields'):
+            # Drop any fields that are specified in the `anonymized_fields` variable.
+            not_allowed = self.anonymized_fields
 
         fields = [
             field for field in self.fields.values() if
-            not field.write_only and field.field_name not in to_be_removed
+            not field.write_only and field.field_name not in not_allowed
```

Comment:
Do we not have a `HideIfAnonymous` field composition? (search search) I guess not. We did make the VOL code a while back, before we were way into field compositions.

Reply:
I can go this route! Do we have the time?

Reply:
Yeah, definitely not for right now.  I was just surprised until I went through more and remembered how/why we went this route. Reticketing might be a good plan, though, especially if we have this problem again, just to keep things consistent. If we did that, we might also consider doing the same for filterable fields. Or we just save it for v3.

Reply:
Reticketed! Can you help me with the `Why`? https://openscience.atlassian.net/browse/ENG-445

Reply:
Will do